member-service/src/main/java/com/hedvig/memberservice/services/cashback/LocalizedCashbackService.kt

```diff
@@ -0,0 +1,111 @@
+package com.hedvig.memberservice.services.cashback
+
+import com.hedvig.common.localization.LocalizationService
+import com.hedvig.memberservice.aggregates.PickedLocale
+import com.hedvig.memberservice.web.dto.CashbackOption
+import com.hedvig.memberservice.web.dto.NonLocalizedCashbackOption
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Component
+import java.util.*
+
+interface CashbackService {
+    fun getCashbackOption(cashbackId: UUID?, pickedLocale: PickedLocale): Optional<CashbackOption>
+    fun getOptions(pickedLocale: PickedLocale): List<CashbackOption>
+    fun getDefaultId(pickedLocale: PickedLocale): UUID
+    fun getDefaultCashback(pickedLocale: PickedLocale): CashbackOption?
+}
+
+@Component
+@ConditionalOnProperty(name = ["localizedCashback"], havingValue = "true")
+class LocalizedCashbackService(
+    private val localizationService: LocalizationService
+) : CashbackService {
+
+    init {
+        println()
+    }
+
+    override fun getCashbackOption(cashbackId: UUID?, pickedLocale: PickedLocale): Optional<CashbackOption> {
+        try {
+            val cashbackOption = norwegianOptions[cashbackId]
+            if (cashbackOption != null) return Optional.ofNullable(localize(cashbackOption, pickedLocale.locale))
+        } catch (ex: NullPointerException) {
+        }
+        return Optional.ofNullable(swedishOptions[cashbackId]?.let { localize(it, pickedLocale.locale) })
+    }
+
+    override fun getOptions(pickedLocale: PickedLocale) =
+        if (isNorwegian(pickedLocale)) {
+            norwegianOptions.values.toList().map { localize(it, pickedLocale.locale) }
+        } else swedishOptions.values.toList().map { localize(it, pickedLocale.locale) }
+
+    override fun getDefaultId(pickedLocale: PickedLocale): UUID = if (isNorwegian(pickedLocale)) {
+        UUID.fromString("02c99ad8-75aa-11ea-bc55-0242ac130003")
+    } else {
+        UUID.fromString("97b2d1d8-af4a-11e7-9b2b-bbc138162bb2")
+    }
+
+    override fun getDefaultCashback(pickedLocale: PickedLocale) =
+        swedishOptions[getDefaultId(pickedLocale)]?.let { localize(it, pickedLocale.locale) }
+
+    private fun localize(nonLocalized: NonLocalizedCashbackOption, locale: Locale) = CashbackOption(
+        nonLocalized.id,
+        localizationService.getTranslation(nonLocalized.nameKey, locale),
+        localizationService.getTranslation(nonLocalized.titleKey, locale),
+        localizationService.getTranslation(nonLocalized.descriptionKey,locale),
+        nonLocalized.selected,
+        nonLocalized.charity,
+        nonLocalized.imageUrl,
+        nonLocalized.selectedUrl,
+        localizationService.getTranslation(nonLocalized.signatureKey, locale),
+        localizationService.getTranslation(nonLocalized.paragraphKey, locale)
+    )
```

Comment:
Is it possible to add the names here?
```suggestion
    private fun localize(nonLocalized: NonLocalizedCashbackOption, locale: Locale) = CashbackOption(
        id = nonLocalized.id,
        ...
    )
```

Reply:
If it's Java it won't work of course